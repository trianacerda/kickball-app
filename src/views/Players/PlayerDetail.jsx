import { useEffect, useState } from 'react';
import { NavLink } from 'react-router-dom';
import { getPlayerById } from '../../services/players';

function PlayerDetail({ match }) {
  const { playerId } = match.params;
  const [player, setPlayer] = useState(null);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    getPlayerById(playerId)
      .then((resp) => setPlayer(resp))
      .finally(() => setLoading(false));
  }, [playerId]);

  if (loading) return <h1>Please wait, your selected Player is loading...</h1>;

  return (
    <>
      <NavLink to="/" className="home-link" exact>
        Home
      </NavLink>
      <NavLink to="/teams" className="team-link" exact>
        Teams
      </NavLink>
      <NavLink to="/players" className="player-link" exact>
        Players
      </NavLink>
      <h1>{player.name}</h1>
      <ul>
        <li>Position: {player.position}</li>
        <li>Team: {player.teams.name}</li>
        <li>
          From: {player.city}, {player.state}
        </li>
      </ul>
    </>
  );
}

export default PlayerDetail;
