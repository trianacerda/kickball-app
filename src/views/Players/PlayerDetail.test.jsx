import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import PlayerDetail from './PlayerDetail';

it('should render a detailed view of an individual player by id ', async () => {
  const { container } = render(
    <MemoryRouter>
      <PlayerDetail match={{ params: { playerId: '1' } }} />
    </MemoryRouter>
  );

  screen.getByText('Please wait, your selected player is loading...');
  const playerName = await screen.findByText('Jonny Clutch');

  expect(playerName).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
