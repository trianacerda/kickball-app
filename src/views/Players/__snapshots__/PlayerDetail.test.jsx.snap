// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a detailed view of an individual player by id  1`] = `
<div>
  <a
    aria-current="page"
    class="home-link active"
    href="/"
  >
    Home
  </a>
  <a
    class="team-link"
    href="/teams"
  >
    Teams
  </a>
  <a
    class="player-link"
    href="/players"
  >
    Players
  </a>
  <h1>
    Jonny  
Clutch
  </h1>
  <ul>
    <li>
      Position: 
      Catcher
    </li>
    <li>
      Team: 
      Tala Loves Snacks
    </li>
    <li>
      From: 
      Sensitive
      , 
      OR
    </li>
  </ul>
</div>
`;
